feat: MiniPlayer 연동

devjaeyoon opened the pull request:
### ✨ 이슈 번호

- #91 

### 📌 설명

라디오 채널에서 재생 버튼을 누르면 MiniPlayer가 나타나도록 연동 작업을 진행한 Pull Request입니다.

### 📃 작업 사항

- [x] MiniPlayer 연동

### 💡 참고 사항

* 이번 PR에서는 MiniPlayer와 ChannelPlayer가 동일한 전역 오디오 자원(video) 를 공유하도록 구조를 통합했습니다.
* useChannelPlayback 훅에 "mini" | "full" 모드를 도입하여, 재생 컨텍스트를 명확히 분리했습니다.
* "full": ChannelPlayer에서 현재 선택된 채널 기준으로 재생 상태를 판단합니다.
* "mini": playingChannelId 기준으로 미니플레이어 재생 상태를 판단합니다.
* A 채널에서 재생 중인 상태로 B 채널 상세로 이동해도, MiniPlayer에는 A 채널 정보 및 재생 상태가 유지됩니다.
* 이때 B 채널에서 재생 버튼을 누르면, MiniPlayer는 B 채널로 업데이트되며 A → B로 재생이 전환됩니다.
* videoRef 대신 전역 video 요소를 사용하도록 변경되었기 때문에, 미니플레이어를 종료할 때 video.pause() 및 video.src = "" 처리로 
명시적 정리 로직을 추가했습니다.

### 💭 리뷰 사항 반영

- [x] `MiniPlayer`와 `ChannelPlayer` 컴포넌트에 `<video>` 태그 제거

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

devjaeyoon committed:
fix: 잘못된 PlayIcon import 경로 수정

devjaeyoon committed:
style: MiniPlayer 디자인 수정

- 하단 border-radius를 상단 border-radius로 변경
- position: absolute 및 bottom: 0으로 하단 고정

devjaeyoon committed:
refactor: isPlaying 상태 분리 및 usePlayingStore 생성

- useChannelStore에서 isPlaying 관련 상태와 로직을 분리
- isPlaying 관리를 위한 usePlayingStore 신규 생성
- useChannelStore는 관심 채널, 이전 채널, 라디오 채널 리스트 등만 관리하도록 변경
- channel-storage로 persist 네이밍 명확화

devjaeyoon committed:
feat: 미니 플레이어 상태를 위한 zustand 스토어 생성

홈/미니 플레이어 전역 상태 관리를 위해 zustand 스토어 생성
- isVisible: 미니 플레이어 표시 여부
- playingChannelId: 재생 중인 채널 ID
- openMiniPlayer, closeMiniPlayer 액션 정의

devjaeyoon committed:
feat: 전역 video element 반환 유틸 함수 getGlobalVideo 추가

- id가 'global-radio'인 video element를 반환하는 getGlobalVideo 함수 생성
- MiniPlayer와 useChannelPlayback에서 전역 video 제어 시 활용

devjaeyoon committed:
refactor: 전역 video 요소를 직접 주입받도록 재생 제어 로직 수정

- videoRef.current 방식 대신 전역 video를 파라미터로 전달받아 사용
- controlStreamingPlayback, controlStreamingSwitch 인터페이스 변경
- MainLayout에서 전역 video를 사용하는 구조에 맞춰 유연성 개선

devjaeyoon committed:
feat: MainLayout에 MiniPlayer 도입 및 전역 video element 추가

- MiniPlayer 컴포넌트 추가 및 isVisible 상태로 표시 여부 제어
- 전역 재생 관리를 위해 video element를 MainLayout에 배치
- useMiniPlayerStore에서 isVisible 상태 사용

devjaeyoon committed:
feat: useChannelPlayback에 mode 분기 도입 및 ChannelPlayer에 full 모드 적용

MiniPlayer와 ChannelPlayer 간 재생 상태 분리를 위해 useChannelPlayback 훅에
mode 파라미터("mini", "full") 추가, mini 모드에서는 playingChannelId 기준으로,
full 모드에서는 selectedChannelId 기준으로 재생 판단
미니 플레이어가 재생 중일 때도 다른 채널 페이지로 이동해 전역 재생 상태와 정보를 유지할 수 있도록 개선
ChannelPlayer 컴포넌트에서 useChannelPlayback("full")로 명시적 모드 전달

devjaeyoon committed:
refactor: MiniPlayer에 전역 상태 및 video 제어 로직 적용

useChannelPlayback("mini")로 모드를 분기하여 MiniPlayer와
ChannelPlayer의 재생 컨텍스트를 분리
getGlobalVideo()를 사용해 전역 video 요소를 제어하도록 변경
handleClose 함수 추가: 재생 중단 후 video src 초기화 및
전역 상태(isPlaying, playingChannelId) 초기화 처리
props로 받던 closePlayer 제거하고 zustand의 closeMiniPlayer()
사용하도록 수정

d0-neii commented on src/hooks/useChannelPlayback.jsx:
```diff
@@ -1,32 +1,46 @@
+    ? isPlaying && playingChannelId !== null
+    : isPlaying && selectedChannelId === playingChannelId;
 
   const handlePlayPause = () => {
```
`handlePlayPause` 안에서 재생 관련 로직이 한꺼번에 처리되고 있어서,
`play / pause`를 함수로 분리하면 가독성이나 유지보수 측면에서 더 좋을 것 같은데

@devjaeyoon 재윤님 생각도 궁금합니다!!

devjaeyoon replied:
현재 handlePlayPause는 play와 pause 조건을 명확하게 분기하고,
해당 시점에 필요한 동작들을 순차적으로 실행하고 있어 응집도가 충분히 높다고 생각합니다.

로직이 복잡하거나 재사용될 필요가 있다면 분리도 고려했겠지만,
현재는 조건이 단순하고 흐름을 한 눈에 파악할 수 있어 그대로 유지하는 쪽이
오히려 가독성과 유지보수 측면에서 이점이 있다고 판단했습니다.

물론, 추후 조건이 늘어나거나 공통화가 필요한 경우 함수 분리도 고려하겠습니다!

faber-h commented on src/layout/MainLayout.jsx:
```diff
@@ -1,15 +1,27 @@
         <main>
           <Outlet />
         </main>
+        <video
```
`video` 태그를 `ChannelPlayer`와 `MiniPlayer` 통합하기 위해 상단으로 잘 끌어올리셨네요!
상위로 끌어올렸음에도 `ChannelPlayer`와 `MiniPlayer`에 video 태그를 남겨놓으신 이유가 있는지 궁금합니다!
`<video ref={videoId} className="hidden" />`

devjaeyoon replied:
좋은 포인트 감사합니다! 말씀 주신 부분 확인해 보니,
현재 `useChannelPlayback`에서 반환하는 `videoId`는
전역 video 태그 (`#global-radio`)를 참조하고 있어
각 컴포넌트 내의 `<video ref={videoId} />`는 실제로 중복된 태그가 되고 있습니다.

말씀대로 `MiniPlayer`와 `ChannelPlayer` 내부의 `<video>`는 제거해도
정상 동작에 문제가 없기 때문에, 해당 부분은 정리해두겠습니다.

devjaeyoon committed:
refactor: MiniPlayer와 ChannelPlayer 컴포넌트에서 불필요한 video 태그 제거

- video 태그는 직접 DOM 접근(getGlobalVideo)을 통해 관리되므로 JSX 내 video 요소 불필요
- 컴포넌트 내 불필요한 요소 제거로 코드 가독성 및 유지보수성 향상

devjaeyoon committed:
refactor: useChannelPlayback 훅 반환값의 순서 정리

- 반환값 순서를 videoId → selectedChannel → isPlaying → handlePlayPause로 변경
- 사용 컴포넌트에서 구조분해 시 일관된 순서 유지 목적

faber-h reviewed (approved):
LGTM🔥

Eun0713 reviewed (approved):
LGTM☀️
